Components/Badge/badge.css

```diff
@@ -0,0 +1,261 @@
+* {
+    box-sizing: border-box;
+    font-family: sans-serif;
+}
+
+:root {
+    /* badge-on-text */
+    --text-badge-bg-color: transparent;
+    --text-badge-border: 1px dashed black;
+    --badge1-bg-color: #bae6fd;
+    --badge1-border: 1px solid black;
+    /* badge-on-button */
+    --button-badge-bg-color: #bae6fd;
+    --button-badge-border: 1px solid black;
+    --badge2-bg-color: #2563eb;
+    --badge2-border: 1px solid black;
+    /* badge-on-icon */
+    --icon-badge-color: #059669;
+    --badge3-bg-color: #dc2626;
+    --badge3-border: 1px solid white;
+    /* badge-on-avatar */
+    --avatar-badge-border: none;
+    --badge4-bg-color: #ff0000;
+    --badge4-border: 1px solid white;
+    /* focus color is seen when you use tab key to access buttons or links*/
+    --badge-focus-outline-color: black;
+}
+
+/* text with badge */
+
+.text-with-badge {
+    width: max-content;
+    position: relative;
+}
+
+.badge-text {
+    background-color: var(--text-badge-bg-color);
+    border: var(--text-badge-border);
+    font-size: 1.7rem;
+    margin: 0;
+    padding: 0.2em 0.4em;
+}
+
+.badge-small .badge-text {
+    font-size: 1rem;
+    padding: 0.2em 0.6em;
+}
+
+.badge-large .badge-text {
+    font-size: 2.5rem;
+}
+
+.badge-badge1 {
+    font-size: 0.63rem;
+    text-align: center;
+    width: 2rem;
+    height: 2rem;
+    background-color: var(--badge1-bg-color);
+    border: var(--badge1-border);
+    border-radius: 50%;
+    position: absolute;
+    right: -6%;
+    top: -18%;
+
+    /* flex-properties */
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.badge-small .badge-badge1 {
+    font-size: 0.4rem;
+    width: 1.3rem;
+    height: 1.3rem;
+    right: -6%;
+    top: -20%;
+}
+
+.badge-large .badge-badge1 {
+    font-size: 1rem;
+    width: 3rem;
+    height: 3rem;
+    right: -6.5%;
+    top: -18%;
+}
+
+/* button with badge */
+
+.button-with-badge {
+    width: max-content;
+    position: relative;
+}
+
+.badge-button {
+    cursor: pointer;
+    font-size: 1.6rem;
+    margin: 0;
+    padding: 0.2em 0.5em;
+    background-color: var(--button-badge-bg-color);
+    border: var(--button-badge-border);
+    border-radius: 0.5em;
+}
+
+.badge-button:focus {
+    outline: var(--badge-focus-outline-color) solid 2px;
+}
+
+.badge-small .badge-button {
+    font-size: 0.9rem;
+    padding: 0.2em 0.7em;
+}
+
+.badge-large .badge-button {
+    font-size: 2.4rem;
+    padding: 0.2em 0.4em;
```

Comment:
You can use padding and margin utility classes, as they will provide you greater flexibility.

Reply:
Yes, will work on it later. Thanks for the review!